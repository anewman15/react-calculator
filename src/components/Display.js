/*
 eslint-disable no-unused-vars
*/

import React from 'react';
import PropTypes from 'prop-types';

const Display = props => (
  <div>{props.calcResult}</div>
);

Display.propTypes = {
  calcResult: PropTypes.string.isRequired,
};

export default Display;