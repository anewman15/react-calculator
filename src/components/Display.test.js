/*
 eslint-disable no-unused-vars, import/named, import/no-cycle
*/

import React from 'react';
import { act, render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import pretty from 'pretty';
import Display from './Display';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Display', () => {
  it('should render a display for the calculated value', () => {
    act(() => {
      render(<Display />, container);

      expect(pretty(container.innerHTML))
        .toMatchSnapshot();
    });
  });
});
