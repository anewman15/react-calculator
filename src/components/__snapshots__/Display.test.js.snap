// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render a display for the calculated value 1`] = `""`;
